back/app/controllers/web_api/v1/insights/phase_insights_controller.rb

```diff
@@ -25,6 +38,28 @@ def set_phase
           @phase = Phase.find params[:phase_id]
           authorize @phase
         end
+
+        def validate_and_find_custom_field(group_by)
+          return nil if group_by.blank?
+
+          custom_field = CustomField.find_by(key: group_by)
+          error = nil
+
+          if custom_field.nil?
+            error = 'custom_field not found with the key provided'
+          elsif custom_field.resource_type != 'User'
+            error = 'Invalid custom_field resource_type for grouping'
+          elsif !custom_field.support_reference_distribution? && custom_field.key != 'birthyear'
+            error = 'Custom field input_type or key not supported for grouping'
+          end
+
+          if error
+            render json: { errors: { group_by: [{ error: error }] } }, status: :unprocessable_entity
+            return nil
+          end
```

Comment:
Think it would be nicer to just return an error from this method and do the rendering of the error in and if else block in the end_point itself. Also combine this with the other error state `@phase.participation_method == 'voting'` so there is only one rendering of error.